pandaharvester/harvesterscripts/prescript.py

```diff
@@ -0,0 +1,41 @@
+import os
+import sys
+import argparse
+
+from pandaharvester.harvesterconfig import harvester_config
+from pandaharvester.harvestermisc.selfcheck import harvesterPackageInfo
+from pandaharvester.harvestercore import fifos
+
+
+def repopulate_fifos():
+    agent_fifo_class_name_tuple = ('MonitorFIFO',)
+    for agent_fifo_class_name in agent_fifo_class_name_tuple:
+        fifo = getattr(fifos, agent_fifo_class_name)()
+        if not fifo.enabled:
+            continue
+        fifo.populate(clear_fifo=True)
+        print('Repopulated {0} fifo'.format(fifo.agentName))
+
+
+def main():
+    oparser = argparse.ArgumentParser(prog='prescript', add_help=True)
+    oparser.add_argument('-f', '--local_info_file', action='store', dest='local_info_file', help='path of harvester local info file')
+
+    if len(sys.argv) == 1:
+        print('No argument or flag specified. Did nothing')
+        sys.exit(0)
+    args = oparser.parse_args(sys.argv[1:])
+
+    local_info_file = os.path.normpath(args.local_info_file)
+
+    hpi = harvesterPackageInfo(local_info_file=local_info_file)
+    if hpi.package_changed:
```

Comment:
Isn't it inefficient to repopulate FIFO every time commit_timestamp is updated? Also how do you protect redundant repopulation if multiple harvester instances are running with the same database? In the first place, do you still need to repopulate FIFO even after you add the change in spec_base which unpickles old object to new object which is pushed to FIFO?

Reply:
Moved out fifo repoputlation from prescript